### ISSUE TITLE
[Bug]: Can't build on Linux

### ISSUE THREAD

**Issue by Pysis868** [obs-ndi/obs-ndi#932]
### Operating System Version

See additional context.

### OBS Version

None

### NDI Tools Version

None

### Describe the bug

See issue title and steps.

### Steps to reproduce

1. Install `zsh`
2. cd `.../obs-ndi`
3. `.github/scripts/build-linux`

Actual behavior:
`.github/scripts/build-linux: line 1: .build.zsh: command not found`

Also:
1. Install `zsh`
2. cd `.../obs-ndi/.github/scripts`
3. `./build-linux`

Actual behavior:
`./build-linux: line 1: .build.zsh: command not found`

Also:
1. Install `zsh`
2. cd `.../obs-ndi/.github/scripts`
3. Change `.github/scripts/build-linux` to `./.build.zsh`.
5. `./build-linux`
 
Actual behavior:
(with additional investigative echo statements):
```
ZSH_ARGZERO: ./.build.zsh
host_os: b
build:139: check_b: function definition file not found
```

### Expected behavior

'...'
Continuing following the README > Building > Linux steps.

### Additional context

> sed -n (read)'p' (read)
read> 139
read> .build.zsh
      -c|--config)

> git log -1 --oneline
bdd6e26 (HEAD -> master, tag: 4.13.0) Merge pull request #926 from obs-ndi/4.13.0_OBS30

> neofetch --stdout distro kernel shell
distro: Fedora release 39 (Thirty Nine) x86_64 
kernel: 6.5.11-300.fc39.x86_64 
shell: fish 3.6.1

Similar enough Linux platform.
No unique errors found yet.
Seems like script/logic errors.
I don't see other issues failing this early.

Somewhat related issues:
- https://github.com/obs-ndi/obs-ndi/issues/698
- https://github.com/obs-ndi/obs-ndi/issues/311
- https://github.com/obs-ndi/obs-ndi/discussions/646
- https://github.com/obs-ndi/obs-ndi/pull/299
- https://github.com/obs-ndi/obs-ndi/issues/137

**Comment by paulpv** [comment]
@Pysis868 Please do this:
```
$ cd obs-ndi
$ ls -la .github/scripts
```

And paste the results here.

I think your chmod or symlinks are missing.

**Comment by tytan652** [comment]
Those scripts are not meant to be used outside of CI and are Ubuntu only, this is why those are hidden in the workflow folder.

**Comment by paulpv** [comment]
> Cloned on another filesystem that wasn't FAT32 now, ext4. Files seem to be linked better.

Correct; the symlinks must be valid, and in your first listing they looked not to be.
This is just a copy of that OBS does and limited to their solution.
The symlinks also break in Windows checkouts, but that is normally not an issue since these links are never run...
...unless you are doing Linux builds under WSL.
It is too easy to accidentally mess up, and I have at least once (and took several force pushes to fix the CI).

> Now display this message:
> .github/scripts/build-linux
   =>   Check Linux build requirements
   ✖︎   No apt-get command found. Please install apt
> So just the OS restriction then.

Yup. Just install apt if you can.
If you cannot then, yes, it is an OS restriction.
Again, inherited from OBS and I would prefer to not stray from OBS plugin template too much (so that when they fix/improve things they are easier to cherry-pick in).

The most important line from our readme relevant to this might be:
> ## Linux
> NOTE: Only Debian and Ubuntu are officially supported

That is my quick paraphrase of the same restriction that OBS has...
https://github.com/obsproject/obs-studio/wiki/Install-Instructions#linux
...although my statement may not be 100% true/precise/accurate.
If you have suggestions on rewording or reformatting then please do provide them.

Think we can we close this Issue?
If you want to be able to build on a currently unsupported platform, feel free to open a "Feature Request" type of new Issue.

I would love to be able to build this on an RPi and/or Jetson Nano.
I tried on an RPi4 awhile ago and gave up.
I am waiting on my RPi5 before I try again.
Main reason is because in my opinion/experience OBS does not run very well on an RPi4, and adding obs-ndi would just make it worse; RPi5 might be better.

**Comment by paulpv** [comment]
Closing this issue; feel free to re-open if you disagree.
